op-challenger/game/fault/test/claim_builder.go

```diff
@@ -31,54 +31,50 @@ func (c *ClaimBuilder) CorrectTraceProvider() types.TraceProvider {
 	return c.correct
 }
 
-// CorrectClaim returns the canonical claim at a specified trace index
-func (c *ClaimBuilder) CorrectClaim(idx uint64) common.Hash {
-	value, err := c.correct.Get(context.Background(), idx)
-	c.require.NoError(err)
-	return value
-}
-
 // CorrectClaimAtPosition returns the canonical claim at a specified position
 func (c *ClaimBuilder) CorrectClaimAtPosition(pos types.Position) common.Hash {
-	value, err := c.correct.Get(context.Background(), pos.TraceIndex(c.maxDepth))
+	value, err := c.correct.Get(context.Background(), pos)
 	c.require.NoError(err)
 	return value
 }
 
 // CorrectPreState returns the pre-state (not hashed) required to execute the valid step at the specified trace index
 func (c *ClaimBuilder) CorrectPreState(idx uint64) []byte {
-	preimage, _, _, err := c.correct.GetStepData(context.Background(), idx)
+	pos := types.NewPosition(c.maxDepth, int(idx))
+	preimage, _, _, err := c.correct.GetStepData(context.Background(), pos)
 	c.require.NoError(err)
 	return preimage
 }
 
 // CorrectProofData returns the proof-data required to execute the valid step at the specified trace index
 func (c *ClaimBuilder) CorrectProofData(idx uint64) []byte {
-	_, proof, _, err := c.correct.GetStepData(context.Background(), idx)
+	pos := types.NewPosition(c.maxDepth, int(idx))
+	_, proof, _, err := c.correct.GetStepData(context.Background(), pos)
 	c.require.NoError(err)
 	return proof
 }
 
 func (c *ClaimBuilder) CorrectOracleData(idx uint64) *types.PreimageOracleData {
-	_, _, data, err := c.correct.GetStepData(context.Background(), idx)
+	pos := types.NewPosition(c.maxDepth, int(idx))
+	_, _, data, err := c.correct.GetStepData(context.Background(), pos)
 	c.require.NoError(err)
 	return data
 }
 
-func (c *ClaimBuilder) incorrectClaim(idx uint64) common.Hash {
-	return common.BigToHash(new(big.Int).SetUint64(idx))
+func (c *ClaimBuilder) incorrectClaim(pos types.Position) common.Hash {
+	return common.BigToHash(new(big.Int).SetUint64(pos.TraceIndex(c.maxDepth)))
 }
 
-func (c *ClaimBuilder) claim(idx uint64, correct bool) common.Hash {
+func (c *ClaimBuilder) claim(pos types.Position, correct bool) common.Hash {
 	if correct {
-		return c.CorrectClaim(idx)
+		return c.CorrectClaimAtPosition(pos)
 	} else {
-		return c.incorrectClaim(idx)
+		return c.incorrectClaim(pos)
 	}
 }
 
 func (c *ClaimBuilder) CreateRootClaim(correct bool) types.Claim {
-	value := c.claim((1<<c.maxDepth)-1, correct)
+	value := c.claim(types.NewPositionFromGIndex((1<<c.maxDepth)-1), correct)
```

Comment:
```suggestion
	value := c.claim(types.NewPositionFromGIndex(1), correct)
```

The root position is always GIndex 1 and its trace index is the right most index.

Reply:
Oh slipped my mind, thanks!